File: Models/Token.cs
namespace financial_manager.Models
{
    public class Token
    {
        public int Id { get; set; }
        public string RefreshToken { get; set; } = null!;
        public bool IsRevoked { get; set; }
        public DateTime ExpirationDate { get; set; }
        public User? User { get; set; }
    }
}

File: financialManagerUI/src/app/entities/models/token.ts
import { BaseEntity } from '../shared/baseEntity';
import { User } from './user';

export class Token extends BaseEntity {
  refreshToken: string;
  expirationDate: Date;
  isRevoked: boolean;
  user?: User = undefined;

  constructor(
    refreshToken: string,
    expirationDate: Date,
    isRevoked: boolean,
    user?: User,
    id?: number
  ) {
    super();
    this.refreshToken = refreshToken;
    this.expirationDate = expirationDate;
    this.isRevoked = isRevoked;
    this.user = user;
    this.id = id;
  }
}
